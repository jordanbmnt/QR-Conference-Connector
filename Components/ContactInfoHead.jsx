import {
  Avatar,
  AvatarFallbackText,
  AvatarImage,
  Box,
  HStack,
  Heading,
  Icon,
} from "@gluestack-ui/themed";
import { Dot } from "lucide-react-native";

export default ContactInfoHead = ({
  profilePic,
  name,
  status,
  statusColor,
}) => {
  return (
    <Box sx={{ alignItems: "center" }}>
      <Avatar size='xl' mb={20}>
        <AvatarFallbackText>{name[0]}</AvatarFallbackText>
        <AvatarImage
          source={{
            uri: profilePic,
          }}
        />
      </Avatar>
      <Heading color='white' size='2xl'>
        {name}
      </Heading>
      {status && (
        <HStack sx={{ right: 10 }}>
          <Icon color={statusColor} as={Dot} size='sm' />
          <Heading color='grey' size='xs'>
            {status}
          </Heading>
        </HStack>
      )}
    </Box>
  );
};
