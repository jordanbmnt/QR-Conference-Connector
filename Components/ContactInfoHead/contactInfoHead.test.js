import renderer from 'react-test-renderer';
import ContactInfoHead from './ContactInfoHead';

describe("ContactInfoHead", function(){
    const contactInfo = renderer.create(
            <ContactInfoHead profilePic="noContact" name="Jake" status="available" statusColor="purple"/>
    ).toJSON();
    it("should have three children", function(){
        expect(contactInfo.children.length).toBe(3);
    });
});