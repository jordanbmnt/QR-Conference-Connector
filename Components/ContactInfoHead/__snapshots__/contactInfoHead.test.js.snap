// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInfoHead should render the contactInfo component correctly 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    mb={20}
    size="xl"
    style={
      [
        undefined,
        {},
        undefined,
      ]
    }
  >
    <Text
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          undefined,
          {},
          undefined,
        ]
      }
    >
      J
    </Text>
    <Image
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      onError={[Function]}
      source={
        {
          "uri": "noContact",
        }
      }
      style={
        [
          undefined,
          {},
          undefined,
        ]
      }
    />
  </View>
  <Text
    accessibilityRole="header"
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    size="2xl"
    style={
      {
        "color": "white",
        "fontSize": 32,
        "fontWeight": "bold",
        "marginVertical": 42.56,
      }
    }
  >
    Jake
  </Text>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {
          "right": 10,
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="sm"
      bbWidth="sm"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      fill="none"
      focusable={false}
      height="sm"
      meetOrSlice={0}
      minX={0}
      minY={0}
      role="image"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          [
            undefined,
            {},
            undefined,
            {},
            {
              "color": "purple",
            },
            undefined,
          ],
          {
            "flex": 0,
            "height": "sm",
            "width": "sm",
          },
        ]
      }
      tintColor="purple"
      vbHeight={24}
      vbWidth={24}
      width="sm"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeLinecap",
            "strokeLinejoin",
          ]
        }
        stroke={
          {
            "type": 2,
          }
        }
        strokeLinecap={1}
        strokeLinejoin={1}
        strokeWidth="2"
      >
        <RNSVGCircle
          cx="12.1"
          cy="12.1"
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          r="1"
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      accessibilityRole="header"
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      size="xs"
      style={
        {
          "color": "grey",
          "fontSize": 32,
          "fontWeight": "bold",
          "marginVertical": 42.56,
        }
      }
    >
      available
    </Text>
  </View>
</View>
`;
